pkg/apis/core/v1beta1/types_imagevector.go

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1beta1
+
+// ImageVector is a list of images.
+type ImageVector []ImageSource
```

Comment:
I started like this (and similarly for `ComponentImageVector`), but abolished the structs at some point. It's just more verbose without any additional value. From my perspective, whatever you can do with a struct type you can do with the slice alias, including "reading the image vector from a file" - there is even already helper code for this, which could be used for what you suggest without any changes.

Reply:
Let's discuss this in person later on, as it might depend on the outcome of our decision regarding https://github.com/gardener/gardener/pull/3656#discussion_r590509303.